File: shared/src/commonMain/kotlin/implementation/assembler/bridging/ViewControllerProtocol.kt
package implementation.assembler.bridging

interface ViewControllerProtocol {
    val nativeNavigationController: NavigationControllerProtocol?

    fun nativePresent(
        viewController: ViewControllerProtocol,
        animated: Boolean,
        completion: (() -> Unit)?,
    )

    fun nativeDismiss(
        animated: Boolean,
        completion: (() -> Unit)?,
    )
}

File: iosApp/iosApp/Bridging/ViewControllerProtocol.swift
//  swiftlint:disable:this file_name
//
//  ViewControllerProtocol.swift
//  iosApp
//

import Foundation
import Shared
import UIKit

extension UIViewController: ViewControllerProtocol {
	public func nativePresent(
		viewController: ViewControllerProtocol,
		animated: Bool,
		completion: (() -> Void)? = nil
	) {
		guard let viewController = viewController as? UIViewController else { return }
		present(viewController, animated: animated)
		completion?()
	}
	
	public var nativeNavigationController: NavigationControllerProtocol? {
		navigationController
	}
	
	public func nativeDismiss(animated: Bool, completion: (() -> Void)? = nil) {
		dismiss(animated: animated, completion: completion)
	}
}
